src/generation.rs

```diff
@@ -788,6 +824,21 @@ impl GenerationScope {
                     }
                 }
             }
+        }
+        // imports and write to file
+        for (scope, content) in self.rust_scopes.iter_mut() {
+            add_imports_from_scope_refs(scope, content, &rust_imports);
+            // TODO: we blindly add these two map imports. Ideally we would only do it when needed
```

Comment:
This isn't that bad I think. Probably we can use something like rustfmt to just delete unused imports after the codegen runs